packages/rum-core/src/performance-monitoring/bootstrap.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+export let state = {}
+
+export function bootstrap() {
+  if (document.visibilityState === 'hidden') {
+    state.lastHidden = {
+      start: 0
+    }
+  }
+
+  window.addEventListener(
+    'visibilitychange',
+    () => {
+      let { lastHidden } = state
+      if (document.visibilityState === 'hidden') {
+        state.lastHidden = {
+          start: performance.now()
```

Comment:
IMO, we can just use this time and rely on it to determine if the page was ever backgrounded and discard the transaction. Do you see any problems with this?

Reply:
I'm not sure what exactly do you mean with this, can you elaborate?

Reply:
I forgot what it was, been a long time 😅